src/linear_algebra/basis.lean

```diff
@@ -397,6 +397,10 @@ by rw [b.constr_def S f, linear_map.range_comp, linear_map.range_comp, linear_eq
        ← finsupp.supported_univ, finsupp.lmap_domain_supported, ←set.image_univ,
        ← finsupp.span_eq_map_total, set.image_id]
 
+@[simp]
+lemma constr_comp (f : M →ₗ[R] M) (v : ι → M) : b.constr ℕ (f ∘ v) = f.comp (b.constr ℕ v) :=
+b.ext (λ i, by simp only [basis.constr_basis, linear_map.comp_apply])
```

Comment:
Is this only true for `ℕ`? You don't seem to use anything special about `ℕ` in the proof

Reply:
Good point, I recall an issue where `simp` didn't work correctly where `S` is a variable, but that doesn't seem to apply now.